missing argument to ‘-fsanitize=’ if USE_SANITIZER is left empty

This seems to have popped up after #408.

In cases where `USE_SANITIZER` is empty (its default value), `add_compile_options("-fsanitize=${san}")` is called with an empty `${san}`.

```
$ colcon build --packages-up-to cyclonedds
Starting >>> cyclonedds
--- stderr: cyclonedds                            
You have called ADD_LIBRARY for library ddsc without any source files. This typically indicates a problem with your CMakeLists.txt file
cc: error: missing argument to ‘-fsanitize=’
cc: error: missing argument to ‘-fsanitize=’
cc: error: missing argument to ‘-fsanitize=’
cc: error: missing argument to ‘-fsanitize=’
cc: error: missing argument to ‘-fsanitize=’
cc: error: missing argument to ‘-fsanitize=’
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_udp.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
cc: error: missing argument to ‘-fsanitize=’
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_raweth.c.o] Error 1
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_ssl.c.o] Error 1
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_tran.c.o] Error 1
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_tcp.c.o] Error 1
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_mcgroup.c.o] Error 1
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_eth.c.o] Error 1
cc: error: missing argument to ‘-fsanitize=’
make[2]: *** [src/core/CMakeFiles/ddsc.dir/ddsi/src/ddsi_ipaddr.c.o] Error 1
make[1]: *** [src/core/CMakeFiles/ddsc.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< cyclonedds [ Exited with code 2 ]

Summary: 0 packages finished [4.21s]
  1 package failed: cyclonedds
  1 package had stderr output: cyclonedds
```

Removing `STATUS` from the `message()` call so that it shows up through `colcon` confirms this

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2811e3a..8a435d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -174,7 +174,7 @@ endif()
 # Make it easy to enable Clang's/gcc's analyzers
 set(USE_SANITIZER "" CACHE STRING "Sanitizers to enable on the build.")
 foreach(san "${USE_SANITIZER}")
-  message(STATUS "Enabling sanitizer: '${san}'")
+  message("Enabling sanitizer: '${san}'")
 
   if("${san}" STREQUAL address)
     add_compile_options(-fno-omit-frame-pointer)
```

```
...
Enabling sanitizer: ''
...
```

I quickly took a look at your CI config, and it seems like you do define `USE_SANITIZER` (to `"none"`), which would explain why this wasn't caught.

I'm building `cyclonedds` @ `master` through `colcon` on Ubuntu 18.04.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,19 +173,21 @@ endif()
 
 # Make it easy to enable Clang's/gcc's analyzers
 set(USE_SANITIZER "" CACHE STRING "Sanitizers to enable on the build.")
-foreach(san "${USE_SANITIZER}")
-  message(STATUS "Enabling sanitizer: '${san}'")
+if(NOT("${USE_SANITIZER}" STREQUAL ""))
+  foreach(san "${USE_SANITIZER}")
+    message(STATUS "Enabling sanitizer: '${san}'")
 
-  if("${san}" STREQUAL address)
-    add_compile_options(-fno-omit-frame-pointer)
-    link_libraries(-fno-omit-frame-pointer)
-  endif()
+    if("${san}" STREQUAL address)
+      add_compile_options(-fno-omit-frame-pointer)
+      link_libraries(-fno-omit-frame-pointer)
+    endif()
 
-  if(NOT("${san}" STREQUAL "none"))
-    add_compile_options("-fsanitize=${san}")
-    link_libraries("-fsanitize=${san}")
-  endif()
-endforeach()
+    if(NOT("${san}" STREQUAL "none"))
+      add_compile_options("-fsanitize=${san}")
+      link_libraries("-fsanitize=${san}")
+    endif()
+  endforeach()
+endif()
 
 
 include(GNUInstallDirs)

